skfem/autodiff/__init__.py

```diff
@@ -0,0 +1,71 @@
+from skfem import DiscreteField, BilinearForm, LinearForm
+from skfem.assembly.form import Form
+from jax import jvp
+from jax.config import config
+from numpy import ndarray
+
+
+config.update("jax_enable_x64", True)
+
+
+class NonlinearForm(Form):
+
+    def linearize(self, u0):
+
+        if not isinstance(u0, DiscreteField):
+            raise NotImplementedError("linearize requires the point around "
+                                      "which the form is linearized as "
+                                      "an argument.")
+
+        @BilinearForm
+        def DF(u, v, w):
+            F1 = lambda U: self.form(U, v, w)
+            return jvp(F1, (u0.astuple,), (u.astuple,))[1]
+
+        @LinearForm
+        def y(v, w):
+            return -self.form(u0.astuple, v, w)
```

Comment:
Although for nonlinear problems one always ends up solving J * du = -f, it may be a good idea to keep this at the level of solving? And return `self.form()` instead of `-self.form()`

Reply:
Sure, I think you have a better overview of this area and I trust your judgment.

Reply:
I agree on that.